src/app/_components/PageHeader.tsx

```diff
@@ -8,25 +8,84 @@ import {
   DescriptionText,
 } from '@/components/DescriptionText'
 import { Heading } from '@/components/Heading'
-import { Meta } from '@/components/Meta'
+import { Meta, type MetaDataType } from '@/components/Meta'
+import { NextStaticImage } from '@/components/NextStaticImage'
 import { SectionDivider } from '@/components/SectionDivider'
 
 import { type CTAProps } from '@/types/sharedProps/ctaType'
-import { type ImageProps } from '@/types/sharedProps/imageType'
+import {
+  type StaticImageProps,
+  type ImageProps,
+} from '@/types/sharedProps/imageType'
 
 type TitleProps = {
   children: string
 }
 
-type PageHeaderProps = {
+type SharedProps = {
   title: TitleProps['children']
   description?: DescriptionTextType
-  cta?: CTAProps
-  secondaryCta?: CTAProps
-  image?: ImageProps
-  metaData?: Array<string | null | undefined>
+  cta: CTAProps | [CTAProps, CTAProps?]
+  metaData?: MetaDataType
   isFeatured?: boolean
-  containImageSize?: boolean
+}
+
+type DynamicImageProps = Partial<ImageProps> & {
+  fallback: StaticImageProps
+  objectFit?: 'contain' | 'cover'
+}
+
+type PageHeaderProps = SharedProps & {
+  image:
+    | ({ type: 'static' } & StaticImageProps)
+    | ({ type: 'dynamic' } & DynamicImageProps)
+}
+
+const imageContainerStyle = 'aspect-video lg:aspect-auto lg:w-1/2'
+
+function StaticImage({ data, alt }: StaticImageProps) {
+  return (
+    <div className={imageContainerStyle}>
+      <NextStaticImage
+        priority
+        data={data}
+        alt={alt}
+        quality={100}
+        className="h-full w-full rounded-lg border border-brand-100 object-cover"
+      />
+    </div>
+  )
+}
+
+function DynamicImage({
+  url,
+  alt,
+  objectFit = 'cover',
+  fallback,
+  ...rest
+}: DynamicImageProps) {
+  if (!url) {
+    return <StaticImage {...fallback} />
+  }
```

Comment:
From Next.js perspective, both static and dynamic images rely on the native `Image` component, but how to use this component differs depending on the kind of image.

With [static images](https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images), the `src` prop accepts a config object containing data about the image's source, width, height, placeholder, etc.

With [dynamic images](https://nextjs.org/docs/app/building-your-application/optimizing/images#remote-images), the `src` prop accepts a URL string, while width, height, placeholder, etc need to be configured manually.

With 2 components, it's easier to ensure that each prop API fits the use case. For instance, `NextStaticImage` does not accept a `width` API because it's computed by Next for us. But `Image` does because it's controlled by us.

Does it make sense?

Reply:
Very cool! ❤️